packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -138,7 +196,11 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
       // Act on the target log group
       await createLogGroupSafe(logGroupName, logGroupRegion, retryOptions);
       await setRetentionPolicy(logGroupName, logGroupRegion, retryOptions, parseInt(event.ResourceProperties.RetentionInDays, 10));
+      if (event.ResourceProperties.PropagateTags === 'true') {
```

Comment:
We also need to handle the `event.RequestType === 'Delete`.

Here is a scenario:
I create a Lambda Function and set propagate tags
```ts
new Function(this, 'Function', { propagateTags: true });
```

Later I decide that I don't want to propagate tags anymore.
```ts
new Function(this, 'Function', { propagateTags: false });
```

This will remove the log retention custom resource which will delete the log
group.

Honestly I think this might also be a bug in the existing implementation. We
should only be destroying the log group if the lambda function is being
destroyed.

Reply:
I think this is covered by the following code:
https://github.com/aws/aws-cdk/blob/5f0d7c8d86cddb02c74581493f41283d6e4d4bcf/packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts#L218-L222
The log group will only be deleted if a delete request is made and the `RemovalPolicy` is destroy. I created the scenario you mentioned and the `LogGroup` was retained.